connectors/spark/src/test/java/io/unitycatalog/spark/ApiClientFactoryTest.java

```diff
@@ -7,7 +7,112 @@
 import java.net.URI;
 import org.junit.jupiter.api.Test;
 
+/**
+ * Test class for ApiClientFactory to verify User-Agent configuration and client setup.
+ */
 public class ApiClientFactoryTest {
+
+  @Test
+  public void testUserAgentContainsSpark() throws Exception {
+    ApiClientConf clientConf = new ApiClientConf();
+    URI uri = new URI("http://localhost:8080");
+    ApiClient client = ApiClientFactory.createApiClient(clientConf, uri, null);
+
+    String userAgent = client.getUserAgent();
+
+    // Verify the user agent contains the base Unity Catalog client
+    assertThat(userAgent).startsWith("UnityCatalog-Java-Client/");
+
+    // Verify the user agent contains Spark (required)
+    assertThat(userAgent).contains("Spark");
+
+    // Extract and verify Spark version (version may be empty but Spark must be present)
+    boolean foundSpark = false;
+    String[] parts = userAgent.split(" ");
+    for (String part : parts) {
+      if (part.startsWith("Spark")) {
+        foundSpark = true;
+        break;
+      }
+    }
+    assertThat(foundSpark).as("Spark must be present in User-Agent").isTrue();
+
+    // Delta is optional - verify it if present
+    boolean foundDelta = false;
+    for (String part : parts) {
+      if (part.startsWith("Delta")) {
+        foundDelta = true;
+        break;
+      }
+    }
+    // Note: Delta may or may not be present depending on classpath
+  }
+
+  @Test
+  public void testUserAgentWithToken() throws Exception {
+    ApiClientConf clientConf = new ApiClientConf();
+    URI uri = new URI("http://localhost:8080");
+    String token = "test-token-12345";
+    ApiClient client = createApiClient(clientConf, uri, token);
+
+    String userAgent = client.getUserAgent();
+
+    // Verify user agent is set correctly even with authentication
+    assertThat(userAgent).startsWith("UnityCatalog-Java-Client/");
+
+    // Verify Spark is present (required)
+    assertThat(userAgent).contains("Spark");
+
+    // Delta is optional
+    // assertThat(userAgent).contains("Delta"); // May or may not be present
```

Comment:
uncomment this, its expected Delta will be present here because this project is compiling with Delta. but it may not be present in a real deployment. that is hard to test in a unit test.